core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -104,22 +104,33 @@ public static WeekActivity createInstance(UserAnonymized userAnonymized, Goal go
 	public static WeekActivity createInstanceInactivity(UserAnonymized userAnonymized, Goal goal, ZonedDateTime startOfWeek)
 	{
 		WeekActivity result = createInstance(userAnonymized, goal, startOfWeek);
-		// add the days
-		// notice this doesn't take care of user time zone changes during the week
-		// so for consistency it is important that the batch script adding inactivity does so
-		for (int i = 0; i < 7; i++)
+		result.addInactivityDays();
+		return result;
+	}
+
+	public void addInactivityDays() 
+	{
+		// if the batch job has already run, skip
+		if(dayActivities.size() < 7)
 		{
-			ZonedDateTime startOfDay = startOfWeek.plusDays(i);
-			if (isInFuture(startOfDay, startOfWeek.getZone()))
+			// notice this doesn't take care of user time zone changes during the week
+			// so for consistency it is important that the batch script adding inactivity does so
+			for (int i = 0; i < 7; i++)
 			{
-				break;
-			}
-			if (goal.wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
-			{
-				result.addDayActivity(DayActivity.createInstanceInactivity(userAnonymized, goal, startOfDay));
+				ZonedDateTime startOfDay = getStartTime().plusDays(i);
+				if (isInFuture(startOfDay, getStartTime().getZone()))
+				{
+					break;
+				}
+				if (getGoal().wasActiveAtInterval(startOfDay, ChronoUnit.DAYS))
+				{
+					if(!dayActivities.stream().anyMatch(dayActivity -> dayActivity.getDate().getDayOfWeek().equals(startOfDay.getDayOfWeek())))
```

Comment:
I don't think it will make it much simpler. If there was no activity on Sunday, Monday will be the first item in the list -- unless we add Sunday on the fly at that moment. If you think that last option should be chosen, I have nothing against it but it sounds as making it more complex.